src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,196 @@
+     * or cannot be opened for any other reason.
+     */
+
+    public void save(ArrayList<Task> taskList) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+
+        for (Task task : taskList) {
+            fw.write(task.toString() + "\n");
+        }
+
+        fw.flush();
+        fw.close();
+    }
+
+    /**
+     * Loads data from data file.
+     * If the data file does not exist, a new data file will be created.
+     *
+     * @return an arraylist of tasks which contains any task read from the data file.
+     * @throws IOException  if the named file exists but is a directory rather than a regular file,
+     * does not exist but cannot be created, or cannot be opened for any other reason.
+     * @throws DataFileChangedException if the data file was changed and any entry contains a wrong format.
+     */
+    public ArrayList<Task> load() throws IOException, DataFileChangedException {
+        File file = new File(filePath);
+
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        // if file not present it creates a file, else it does nothing
+        if (file.createNewFile()) {
+            // exit method if a new file is created
+            return taskList;
+        }
+
+        Scanner sc = new Scanner(file);
+
+        while (sc.hasNext()) {
+            String nextCommand = sc.nextLine();
+            Task task;
+
+            switch (nextCommand.charAt(1)) {
+            case 'D':
+                try {
+                    task = extractDeadline(nextCommand.substring(7)); // [D][X] something by time
+                    break;
+                } catch (DataFileChangedException e) {
+                    throw new DataFileChangedException();
+                }
+            case 'E':
+                try {
+                    task = extractEvent(nextCommand.substring(7)); // [D][X] something at time
+                    break;
+                } catch (DataFileChangedException e) {
+                    throw new DataFileChangedException();
+                }
+            case 'T': // todos
+                task = new Todo(nextCommand.substring(7)); // disregards [T][X]
+                break;
+            default:
+                throw new DataFileChangedException();
+            }
+
+            // check if marked as done
+            if (nextCommand.charAt(4) == 'X') {
+                task.markAsDone();
+            }
+
+            taskList.add(task);
+        }
+
+        sc.close();
+
+        return taskList;
+    }
+
+    /**
+     * Extracts a deadline from the data file in the proper format for duke to read.
+     *
+     * @param text the deadline in the data file.
+     * @return a new deadline that represents the deadline from the data file.
+     * @throws DataFileChangedException if the data file was changed and any entry contains a wrong format.
+     */
+    private Deadline extractDeadline(String text) throws DataFileChangedException {
```

Comment:
Just curious to know. What's the reason for placing this function in the Storage class (rather than Deadline)?